src/main/java/graphql/language/AstPrinter.java

```diff
@@ -523,7 +547,35 @@ private <T extends Node> String join(List<T> nodes, String delim) {
         return join(nodes, delim, "", "");
     }
 
+    /*
+     * Some joined nodes don't need delimiters between them and some do
+     * This encodes that knowledge of those that don't require delimiters
+     */
     @SuppressWarnings("SameParameterValue")
+    private <T extends Node> String jointight(List<T> nodes, String delim, String prefix, String suffix) {
```

Comment:
Hyper nit: could you make it camelCase `joinTight`

For people with poor eyesight like me